src/cdomain/value/domains/valueDomainQueries.ml

```diff
@@ -61,9 +61,7 @@ let eval_int_binop (module Bool: Lattice.S with type t = bool) binop (eval_int:
   if ID.is_bot i || ID.is_bot_ikind i then
     Bool.top () (* base returns bot for non-int results, consider unknown *)
   else
-    match ID.to_bool i with
-    | Some b -> b
-    | None -> Bool.top ()
+    BatOption.default (Bool.top ()) (ID.to_bool i)
```

Comment:
This might make no difference here, but previously `Bool.top` was only called when actually needed, now it's always called but possibly ignored.
There is `BatOption.default_delayed` for that purpose. Maybe there are other instances where that would be more needed.

Reply:
I tried to use it where functions seemed expensive. Given it is a constant here, I would expect `flambda` to inline the boolean (but I may be wrong).